gslib/tests/test_rsync.py

```diff
@@ -1837,25 +1928,52 @@ def _Check1():
-      self.assertTrue(BuildErrorRegex(obj1, POSIX_MODE_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj2, POSIX_GID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj6, POSIX_GID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj7, POSIX_MODE_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj8, POSIX_UID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj9, POSIX_UID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj10, POSIX_UID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj11, POSIX_UID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj12, POSIX_UID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj13, POSIX_UID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj14, POSIX_GID_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj15, POSIX_GID_ERROR).search(stderr))
-      self.assertTrue(
-          BuildErrorRegex(obj16,
-                          POSIX_INSUFFICIENT_ACCESS_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj17, POSIX_MODE_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj18, POSIX_MODE_ERROR).search(stderr))
-      self.assertTrue(BuildErrorRegex(obj19, POSIX_MODE_ERROR).search(stderr))
+      if self._use_gcloud_storage:
+        gcloud_preserve_posix_warning = (
+            r'For preserving POSIX with rsync downloads, gsutil aborts if a single'
+            r' download will result in invalid destination POSIX. However, this'
+        )
+        self.assertRegex(stderr, gcloud_preserve_posix_warning)
+
+        read_regex = r"{}#\d+\. User \d+ owns file, but owner does not have read"
+        gid_regex = r"{}#\d+ metadata doesn't exist on current system\. GID"
+        uid_regex = r"{}#\d+ metadata doesn't exist on current system\. UID"
+        self.assertRegex(stderr, read_regex.format('obj1'))
+        self.assertRegex(stderr, gid_regex.format('obj2'))
+        self.assertRegex(stderr, gid_regex.format('obj6'))
+        self.assertRegex(stderr, read_regex.format('obj7'))
+        self.assertRegex(stderr, uid_regex.format('obj8'))
+        self.assertRegex(stderr, uid_regex.format('obj9'))
+        self.assertRegex(stderr, uid_regex.format('obj10'))
+        self.assertRegex(stderr, uid_regex.format('obj11'))
+        self.assertRegex(stderr, uid_regex.format('obj12'))
+        self.assertRegex(stderr, uid_regex.format('obj13'))
+        self.assertRegex(stderr, gid_regex.format('obj14'))
+        self.assertRegex(stderr, gid_regex.format('obj15'))
+        self.assertRegex(stderr, read_regex.format('obj16'))
+        self.assertRegex(stderr, read_regex.format('obj17'))
+        self.assertRegex(stderr, read_regex.format('obj18'))
+        self.assertRegex(stderr, read_regex.format('obj19'))
+      else:
+        self.assertIn(ORPHANED_FILE, stderr)
+        self.assertTrue(BuildErrorRegex(obj1, POSIX_MODE_ERROR).search(stderr))
```

Comment:
These could also probably be self.assertRegex (though I see that this was gsutil code)

Reply:
Done